### PR TITLE
fix: update dag-ucan, types and names

### PR DESCRIPTION
**Pull request by hugomrdias**
depends on https://github.com/ipld/js-dag-ucan/pull/55

### CODE REVIEW

File: packages/core/src/link.js [hunk 1]
```diff
@@ -5,37 +5,37 @@ import * as API from '@ucanto/interface'
  * @template {number} Code
  * @template {number} Alg
  * @param {Code} code
- * @param {API.MultihashDigest<Alg>} digest
- * @return {API.Link<unknown, Code, Alg, 1> & CID}
+ * @param {API.UCAN.MultihashDigest<Alg>} digest
+ * @return {API.Link<any, Alg>}
```
Comment (Gozala) [inline packages/core/src/link.js:9]: We should not introduce `any`, here it's really `unknown`.

```suggestion
 * @return {API.Link<unknown, Alg>}
```

File: packages/core/src/link.js [hunk 2]
```diff
@@ -5,37 +5,37 @@ import * as API from '@ucanto/interface'
...
  */
 export const create = (code, digest) =>
   /** @type {any} */ (CID.createV1(code, digest))
 
 /**
  * @template {number} Alg
- * @param {API.MultihashDigest<Alg>} digest
- * @return {API.Link<unknown, 0x70, Alg, 0> & CID}
+ * @param {API.UCAN.MultihashDigest<Alg>} digest
+ * @return {API.Link<any, Alg>}
```
Comment (Gozala) [inline packages/core/src/link.js:17]:
```suggestion
 * @return {API.Link<unknown, Alg>}
```

File: packages/core/src/link.js [hunk 3]
```diff
@@ -5,37 +5,37 @@ import * as API from '@ucanto/interface'
...
  */
 export const createV0 = (digest) => /** @type {any} */ (CID.createV0(digest))
 
 /**
  * Type predicate returns true if value is the link.
  *
- * @template {API.Link<unknown, number, number, 0|1>} L
+ * @template {API.Link<any, number>} L
```
Comment (Gozala) [inline packages/core/src/link.js:24]:
```suggestion
 * @template {API.Link<unknown, number>} L
```

File: packages/core/src/link.js [hunk 4]
```diff
@@ -5,37 +5,37 @@ import * as API from '@ucanto/interface'
...
  * @param {unknown} value
- * @returns {value is (L & CID)}
+ * @returns {value is (L)}
```
Comment (Gozala) [inline packages/core/src/link.js:26]:
```suggestion
 * @returns {value is L}
```

File: packages/core/src/link.js [hunk 5]
```diff
@@ -5,37 +5,37 @@ import * as API from '@ucanto/interface'
...
  */
 export const isLink = (value) =>
   value != null && /** @type {{asCID: unknown}} */ (value).asCID === value
 
 export const asLink =
-  /** @type {<L extends API.Link<unknown, number, number, 0|1>>(value:L|unknown) => (L & CID)|null} */
+  /** @type {<L extends API.Link<any, number>>(value:L|unknown) => (L)|null} */
```
Comment (Gozala) [inline packages/core/src/link.js:32]:
```suggestion
  /** @type {<L extends API.Link<unknown, number>>(value:L|unknown) => (L)|null} */
```

File: packages/core/src/link.js [hunk 6]
```diff
@@ -5,37 +5,37 @@ import * as API from '@ucanto/interface'
...
   (CID.asCID)
 
 export const parse =
-  /** @type {<P extends string>(source:string, base?:API.MultibaseDecoder<P>) => API.Link<unknown, number, number, 0|1> & CID} */
+  /** @type {<P extends string>(source:string, base?:API.MultibaseDecoder<P>) => API.Link<any, number>} */
```
Comment (Gozala) [inline packages/core/src/link.js:36]:
```suggestion
  /** @type {<P extends string>(source:string, base?:API.MultibaseDecoder<P>) => API.Link<unknown, number>} */
```

File: packages/core/src/link.js [hunk 7]
```diff
@@ -5,37 +5,37 @@ import * as API from '@ucanto/interface'
...
   (CID.parse)
 
 export const decode =
-  /** @type {(bytes:Uint8Array) => API.Link<unknown, number, number, 0|1> & CID} */
-  CID.decode
+  /** @type {(bytes:Uint8Array) => API.Link<any, number>} */
```
Comment (Gozala) [inline packages/core/src/link.js:40]:
```suggestion
  /** @type {(bytes:Uint8Array) => API.Link<unknown, number>} */
```

File: packages/interface/package.json [hunk 8]
```diff
@@ -23,11 +23,11 @@
     "build": "tsc --build"
   },
   "dependencies": {
-    "@ipld/dag-ucan": "^1.7.0-beta",
-    "multiformats": "^9.6.4"
+    "@ipld/dag-ucan": "link:/Users/hd/code/js-dag-ucan",
```
Comment (Gozala) [inline packages/interface/package.json:26]: should not be a local link

File: packages/interface/src/authority.ts [hunk 9]
```diff
@@ -1,26 +1,21 @@
...
-export interface Authority<A extends number = number>
+export interface Principal<A extends number = number>
   extends ArrayBufferView,
-    UCAN.Authority<A> {
+    UCAN.Verifier<A> {
   bytes: Uint8Array
 }
 
-export interface SigningAuthority<A extends number = number>
+export interface Agent<A extends number = number>
```
Comment (Gozala) [inline packages/interface/src/authority.ts:14]: Please don't get mad for going flip flopping on this, but I do have second thoughts on `Agent` which really should be what we call `Client` today.

That said I think I have good plan here which is we could instead do following:

```ts
export interface Verifier<A extends number = number>
  extends ArrayBufferView,
    UCAN.Verifier<A> {
  bytes: Uint8Array
}

export interface Signer<A extends number = number> extends ArrayBufferView,
    UCAN.Verifier<A>,
    UCAN.Signer<A> {
  verifier: Verifier<A>
  bytes: Uint8Array
  // secret: Uint8Array
}

export type Principal<A> =
  | Signer<A>
  | Verifier<A>
```

That way when we can refer to arbitrary principal as `Principal` but when we need to be specific we could say `Principal.Signer`.

File: packages/transport/src/car/codec.js [hunk 10]
```diff
@@ -110,17 +110,16 @@ export const decode = async (bytes) => {
 
 /**
  * @param {Uint8Array} bytes
- * @param {{hasher?: API.MultihashHasher }} [options]
+ * @param {{hasher?: API.UCAN.MultihashHasher }} options
  */
 export const link = async (bytes, { hasher = sha256 } = {}) =>
-  /** @type {UCAN.Link<Model, typeof code, number> & import('multiformats').CID} */
+  /** @type {UCAN.Link<any, number>} */
```
Comment (Gozala) [inline packages/transport/src/car/codec.js:116]:
```suggestion
  /** @type {UCAN.Link<unknown, number>} */
```

File: packages/transport/src/car/codec.js [hunk 11]
```diff
@@ -110,17 +110,16 @@ export const decode = async (bytes) => {
...
   (createLink(code, await hasher.digest(bytes)))
 
 /**
  * @param {Partial<Model>} data
- * @param {{hasher?: API.MultihashHasher }} [options]
  */
-export const write = async (data, { hasher = sha256 } = {}) => {
+export const write = async (data) => {
```
Comment (Gozala) [inline packages/transport/src/car/codec.js:122]: We should instead pass hasher into a link method

File: packages/transport/src/car/codec.js [hunk 12]
```diff
@@ -110,17 +110,16 @@ export const decode = async (bytes) => {
 
 /**
  * @param {Uint8Array} bytes
- * @param {{hasher?: API.MultihashHasher }} [options]
+ * @param {{hasher?: API.UCAN.MultihashHasher }} options
```
Comment (Gozala) [inline packages/transport/src/car/codec.js:113]: Options should be optional

```suggestion
 * @param {{hasher?: API.UCAN.MultihashHasher }} [options]
```

File: packages/transport/src/cbor/codec.js [hunk 13]
```diff
@@ -78,7 +79,7 @@ export const encode = (data) => CBOR.encode(prepare(data, new Set()))
 /**
  * @template T
  * @param {T} data
- * @param {{hasher?: import('multiformats/hashes/interface').MultihashHasher }} [options]
+ * @param {{hasher?: API.UCAN.MultihashHasher }} options
```
Comment (Gozala) [inline packages/transport/src/cbor/codec.js:82]: Maybe we should export `MultihashHasher` from the API.

File: packages/validator/src/decoder/link.js [hunk 14]
```diff
@@ -11,7 +11,7 @@ export { create, createV0, isLink, asLink, parse }
  * @template {1|0} Version
  * @param {unknown} input
  * @param {{code?:Code, algorithm?:Alg, version?:Version}} [options]
- * @returns {API.Result<API.Link<unknown, Code, Alg, Version>, API.Failure>}
+ * @returns {API.Result<API.Link<any, Alg>, API.Failure>}
```
Comment (Gozala) [inline packages/validator/src/decoder/link.js:14]:
```suggestion
 * @returns {API.Result<API.Link<unknown, Alg>, API.Failure>}
```

File: packages/validator/src/decoder/link.js [hunk 15]
```diff
@@ -43,7 +43,7 @@ export const decode = (input, options = {}) => {
         )
       }
 
-      /** @type {API.Link<unknown, Code, Alg, Version>} */
+      /** @type {API.Link<any, Alg>} */
```
Comment (Gozala) [inline packages/validator/src/decoder/link.js:46]:
```suggestion
      /** @type {API.Link<unknown, Alg>} */
```

File: packages/validator/src/decoder/link.js [hunk 16]
```diff
@@ -56,7 +56,7 @@ export const decode = (input, options = {}) => {
  * @template {number} Alg
  * @template {1|0} Version
  * @param {{code?:Code, algorithm?:Alg, version?:Version}} options
- * @returns {API.Decoder<unknown,  API.Link<unknown, Code, Alg, Version>, API.Failure>}
+ * @returns {API.Decoder<unknown,  API.Link<any, Alg>, API.Failure>}
```
Comment (Gozala) [inline packages/validator/src/decoder/link.js:59]:
```suggestion
 * @returns {API.Decoder<unknown,  API.Link<unknown, Alg>, API.Failure>}
```

File: packages/validator/src/decoder/link.js [hunk 17]
```diff
@@ -68,7 +68,7 @@ export const match = (options) => ({
  * @template {number} Alg
  * @template {1|0} Version
  * @param {{code?:Code, algorithm?:Alg, version?:Version}} [options]
- * @returns {API.Decoder<unknown, undefined|API.Link<unknown, Code, Alg, Version>, API.Failure>}
+ * @returns {API.Decoder<unknown, undefined|API.Link<any,Alg>, API.Failure>}
```
Comment (Gozala) [inline packages/validator/src/decoder/link.js:71]:
```suggestion
 * @returns {API.Decoder<unknown, undefined|API.Link<unknown, Alg>, API.Failure>}
```

File: packages/validator/src/decoder/uri.js [hunk 18]
```diff
@@ -4,7 +4,7 @@ import { Failure } from '../error.js'
 /**
  * @template {`${string}:`} Protocol
  * @param {unknown} input
- * @param {{protocol?: Protocol}} [options]
+ * @param {{protocol?: Protocol}} options
```
Comment (Gozala) [inline packages/validator/src/decoder/uri.js:7]: Options should be optional

```suggestion
 * @param {{protocol?: Protocol}} [options]
```

**Review by Gozala — CHANGES_REQUESTED** [review]

File: packages/authority/package.json [hunk 19]
```diff
@@ -27,20 +27,20 @@
     "build": "tsc --build"
   },
   "dependencies": {
-    "@ipld/dag-ucan": "^1.7.0-beta",
-    "@noble/ed25519": "^1.6.1",
+    "@ipld/dag-ucan": "link:/Users/hd/code/js-dag-ucan",
```
[the remark this answers is not in the document]
Comment (hugomrdias) [reply packages/authority/package.json:30]: need https://github.com/ipld/js-dag-ucan/pull/55 to be release

File: .github/release-please-manifest.json [hunk 20]
```diff
@@ -1 +1,9 @@
-{"packages/authority":"0.5.0","packages/client":"0.6.0","packages/core":"0.6.0","packages/interface":"0.7.0","packages/server":"0.7.0","packages/transport":"0.7.0","packages/validator":"0.6.0"}
```
[the remark this answers is not in the document]
Comment (hugomrdias) [reply .github/release-please-manifest.json:1]: maybe im not familiar with your setup here

File: packages/core/src/link.js (hunk 3 above)
Comment (hugomrdias) [reply packages/core/src/link.js:24]: when used in a generic ts says:

Type 'unknown' does not satisfy the constraint 'Capabilities'.

File: packages/core/src/link.js (hunk 5 above)
Comment (hugomrdias) [reply packages/core/src/link.js:32]: Type 'unknown' does not satisfy the constraint 'Capabilities'.

File: packages/transport/src/car/codec.js (hunk 11 above)
Comment (hugomrdias) [reply packages/transport/src/car/codec.js:122]: did understand this one

File: packages/interface/src/authority.ts (hunk 9 above)
Comment (hugomrdias) [reply packages/interface/src/authority.ts:14]: can we change it in a follow up ?

File: packages/transport/src/car/codec.js (hunk 12 above)
Comment (Gozala) [reply packages/transport/src/car/codec.js:116]: Turns out TS infers it as optional because default is set

**Review by Gozala — APPROVED** [review]

File: packages/server/src/server.js [hunk 21]
```diff
@@ -12,14 +12,14 @@ export {
 /**
  * Creates a connection to a service.
  *
- * @template Service
+ * @template {Record<string, any>} Service
```
Comment (Gozala) [inline packages/server/src/server.js:15]: Hey @hugomrdias I did see these changes in few places, yet I'm not sure why is that necessary. If it is indeed necessary might be a good idea to add a code comment so reading code it would be clear why.

File: packages/interface/src/principal.ts [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+  bytes: Uint8Array
+}
+
+export interface SigningPrincipal<A extends number = number>
+  extends Principal<A>,
+    UCAN.Signer<A> {
+  principal: Principal<A>
+  bytes: Uint8Array
+  // secret: Uint8Array
+}
```
Comment (hugomrdias) [inline packages/interface/src/principal.ts:20]: i think these should extends UCAN.Principal

File: packages/interface/src/transport.ts [hunk 23]
```diff
@@ -57,9 +56,13 @@ export interface HTTPResponse<T = unknown> extends Phantom<T> {
...
+  Alg extends number = number,
+  V extends UCAN.CIDVersion = 1
 > {
-  readonly cid: UCAN.Proof<C[number], A>
-  readonly bytes: UCAN.ByteView<UCAN.UCAN<C[number]>>
+  bytes: ByteView<T>
+  cid: Link<T, Format, Alg, V>
+
+  data?: T
 }
```
Comment (hugomrdias) [inline packages/interface/src/transport.ts:68]: can we just use UCAN.Block

File: packages/interface/src/lib.ts [hunk 24]
```diff
@@ -95,30 +81,29 @@ export interface UCANOptions {
...
 }
 
-export interface Delegation<
-  C extends [Capability, ...Capability[]] = [Capability, ...Capability[]]
-> {
-  readonly root: Transport.Block<C>
-  readonly blocks: Map<string, Transport.Block>
+export interface Delegation<C extends Capabilities = Capabilities> {
+  readonly root: UCANBlock<C>
+  readonly blocks: Map<string, Block>
```
Comment (hugomrdias) [inline packages/interface/src/lib.ts:95]: why can't this be UCAN.Block?

**Review by hugomrdias — COMMENTED** [review]
i really think we should keep the UCAN namespace in the type exports since have a few types with the same name in UCAN and Ucanto

File: packages/interface/src/transport.ts (hunk 23 above)
Comment (hugomrdias) [reply packages/interface/src/transport.ts:68]: if not this should just be IPLD.Block

File: packages/interface/src/lib.ts (hunk 24 above)
Comment (Gozala) [reply packages/interface/src/lib.ts:95]: Because it may contain non UCAN blocks. E.g. when delegation is encoded into a car it may include blocks in capabilities or facts that could be other IPLD things.

Furthermore when we decode invocations from car, we just pass all the blocks into delegation we create even ones that may not be referenced from it and those could include arbitrary things.

Comment (Gozala) [inline packages/interface/src/lib.ts:95]:
```suggestion
  /**
    * Map of all the IPLD blocks that where included with this delegation DAG.
    * Usually this would be blocks corresponding to proofs, however it may
    * also contain other blocks e.g. things that `capabilities` or `facts` may
    * link.
    * It is not guaranteed to include all the blocks of this DAG, as it represents
    * a partial DAG of the delegation desired for transporting.
    *
    * Also note that map may contain blocks that are not part of this
    * delegation DAG. That is because `Delegation` is usually constructed as
    * view / selection over the CAR which may contain bunch of other blocks.
    */
  readonly blocks: Map<string, Block>
```

Comment (Gozala) [reply packages/interface/src/lib.ts:95]: @hugomrdias does this suggested comment helps ?

File: packages/principal/src/signing-principal.js [hunk 25]
```diff
@@ -63,48 +63,48 @@ export const decode = (bytes) => {
...
+ * @param {API.SigningPrincipal<typeof Principal.code>} agent
  * @param {API.MultibaseEncoder<Prefix>} [encoder]
  */
 export const format = ({ bytes }, encoder) =>
   (encoder || base64pad).encode(bytes)
 
 /**
  * @template {string} Prefix
- * @param {string} signingAuthority
+ * @param {string} agent
```
Comment (hugomrdias) [inline packages/principal/src/signing-principal.js:91]:
```suggestion
 * @param {string} principal
```

File: packages/principal/src/signing-principal.js [hunk 26]
```diff
@@ -63,48 +63,48 @@ export const decode = (bytes) => {
...
  * @param {API.MultibaseDecoder<Prefix>} [decoder]
- * @returns {API.SigningAuthority<typeof Authority.code>}
+ * @returns {API.SigningPrincipal<typeof Principal.code>}
  */
-export const parse = (signingAuthority, decoder) =>
-  decode((decoder || base64pad).decode(signingAuthority))
+export const parse = (agent, decoder) =>
```
Comment (hugomrdias) [inline packages/principal/src/signing-principal.js:95]:
```suggestion
export const parse = (principal, decoder) =>
```

File: packages/principal/src/signing-principal.js [hunk 27]
```diff
@@ -63,48 +63,48 @@ export const decode = (bytes) => {
...
+  decode((decoder || base64pad).decode(agent))
```
Comment (hugomrdias) [inline packages/principal/src/signing-principal.js:96]:
```suggestion
  decode((decoder || base64pad).decode(principal))
```

File: packages/principal/src/signing-principal.js [hunk 28]
```diff
@@ -63,48 +63,48 @@ export const decode = (bytes) => {
...
 
 /**
- * @param {API.SigningAuthority<typeof Authority.code>} signingAuthority
+ * @param {API.SigningPrincipal<typeof Principal.code>} agent
```
Comment (hugomrdias) [inline packages/principal/src/signing-principal.js:99]:
```suggestion
 * @param {API.SigningPrincipal<typeof Principal.code>} principal
```

File: packages/principal/test/lib.spec.js [hunk 29]
```diff
@@ -55,101 +56,101 @@ describe('signing authority', () => {
     }
   })
 
-  it('SigningAuthority.decode', async () => {
+  it('Agent.decode', async () => {
```
Comment (hugomrdias) [inline packages/principal/test/lib.spec.js:59]:
```suggestion
  it('SigningPrincipal.decode', async () => {
```

File: packages/principal/test/lib.spec.js [hunk 30]
```diff
@@ -55,101 +56,101 @@ describe('signing authority', () => {
...
 
     assert.throws(
-      () => SigningAuthority.decode(malformed),
+      () => SigningPrincipal.decode(malformed),
       /must contain public key/
     )
   })
 
-  it('SigningAuthority decode encode roundtrip', async () => {
+  it('Agent decode encode roundtrip', async () => {
```
Comment (hugomrdias) [inline packages/principal/test/lib.spec.js:85]:
```suggestion
  it('SigningPrincipal decode encode roundtrip', async () => {
```

File: packages/principal/test/lib.spec.js [hunk 31]
```diff
@@ -55,101 +56,101 @@ describe('signing authority', () => {
...
 
     assert.deepEqual(
-      SigningAuthority.decode(SigningAuthority.encode(signer)),
+      SigningPrincipal.decode(SigningPrincipal.encode(signer)),
       signer
     )
   })
 
-  it('SigningAuthority.format', async () => {
+  it('Agent.format', async () => {
```
Comment (hugomrdias) [inline packages/principal/test/lib.spec.js:94]:
```suggestion
  it('SigningPrincipal.format', async () => {
```

File: packages/principal/test/lib.spec.js [hunk 32]
```diff
@@ -55,101 +56,101 @@ describe('signing authority', () => {
...
 
     assert.deepEqual(
-      SigningAuthority.parse(SigningAuthority.format(signer)),
+      SigningPrincipal.parse(SigningPrincipal.format(signer)),
       signer
     )
   })
 
-  it('SigningAuthority.did', async () => {
+  it('Agent.did', async () => {
```
Comment (hugomrdias) [inline packages/principal/test/lib.spec.js:103]:
```suggestion
  it('SigningPrincipal.did', async () => {
```

File: packages/principal/test/lib.spec.js [hunk 33]
```diff
@@ -55,101 +56,101 @@ describe('signing authority', () => {
...
-      () => Authority.decode(signer.authority.bytes.slice(0, 32)),
+      () => Principal.decode(signer.principal.bytes.slice(0, 32)),
       /Expected Uint8Array with byteLength/
     )
   })
 
-  it('Authority.format', async () => {
-    const signer = await Lib.generate()
+  it('Principal.format', async () => {
+    const agent = await Lib.generate()
```
Comment (hugomrdias) [inline packages/principal/test/lib.spec.js:143]:
```suggestion
    const principal = await Lib.generate()
```

File: packages/principal/test/lib.spec.js [hunk 34]
```diff
@@ -55,101 +56,101 @@ describe('signing authority', () => {
...
 
-    assert.deepEqual(Authority.format(signer.authority), signer.did())
+    assert.deepEqual(Principal.format(agent.principal), agent.did())
```
Comment (hugomrdias) [inline packages/principal/test/lib.spec.js:145]:
```suggestion
    assert.deepEqual(Principal.format(principal.principal), principal.did())
```

File: packages/principal/test/lib.spec.js [hunk 35]
```diff
@@ -55,101 +56,101 @@ describe('signing authority', () => {
...
   })
 
-  it('Authority.encode', async () => {
-    const signer = await Lib.generate()
+  it('Principal.encode', async () => {
+    const agent = await Lib.generate()
```
Comment (hugomrdias) [inline packages/principal/test/lib.spec.js:149]:
```suggestion
    const principal = await Lib.generate()
```

File: packages/principal/test/lib.spec.js [hunk 36]
```diff
@@ -55,101 +56,101 @@ describe('signing authority', () => {
...
 
     assert.deepEqual(
-      [...Authority.encode(signer.authority)],
-      [...signer.authority.bytes]
+      [...Principal.encode(agent.principal)],
+      [...agent.principal.bytes]
```
Comment (hugomrdias) [inline packages/principal/test/lib.spec.js:153]:
```suggestion
      [...Principal.encode(principal.principal)],
      [...principal.principal.bytes]
```
